pr_agent/tools/pr_reviewer.py

```diff
@@ -230,11 +236,13 @@ def _publish_inline_code_comments(self) -> None:
         if get_settings().pr_reviewer.num_code_suggestions == 0:
             return
 
-        review = self.prediction.strip()
+        review_text = self.prediction.strip()
+        review_text = review_text.lstrip('```yaml').rstrip('`')
         try:
-            data = json.loads(review)
-        except json.decoder.JSONDecodeError:
-            data = try_fix_json(review)
+            data = yaml.load(review_text, Loader=SafeLoader)
+        except Exception as e:
+            logging.error(f"Failed to parse AI prediction: {e}")
+            data = try_fix_yaml(review_text)
```

Comment:
**Suggestion:** Consider using `yaml.safe_load()` instead of `yaml.load()` to avoid potential security vulnerabilities associated with loading untrusted YAML files.
```suggestion
        try:
            data = yaml.safe_load(review_text)
        except Exception as e:
            logging.error(f"Failed to parse AI prediction: {e}")
            data = try_fix_yaml(review_text)
```